sql/parser/eval.go

```diff
@@ -70,17 +70,20 @@ func (d DBool) Type() string {
 
 // Compare implements the Datum interface.
 func (d DBool) Compare(other Datum) int {
+	if other == DNull {
+		// NULL compares lesser than any non-NULL value.
+		return 1
+	}
 	v, ok := other.(DBool)
 	if !ok {
-		// Anything other than a DBool compares greater (e.g. `true > NULL`).
-		return 1
+		panic(fmt.Sprintf("unsupported comparison: %s to %s", d.Type(), v.Type()))
```

Comment:
Yeah, we don't actually have any tests that exercise `CASE` statements it seems. I'll add some.